packages/state_queue/lib/src/state_queue.dart

```diff
@@ -169,15 +178,22 @@ abstract class StateQueue<T> extends ValueNotifier<T>
   /// (neither directly nor indirectly), as that makes testing hard as the completion
   /// of the bloc can not be awaited properly.
   @protected
-  void run(StateUpdater<T> updater) {
-    _taskQueue.sink.add(_UpdaterEntry(updater));
+  Future<void> run(StateUpdater<T> updater) async {
```

Comment:
I have been thinking about adding a way to signal to the outside that an operation is done.

But I am not sure a `Future<void>` implemented this way is useful, as this always completes, not matter if there was an error, or if the action was ignored (due to an unexpected state type (e.g. loading) for example). One thing I was pondering with was returning the last state…

Should we maybe for now keep this as an implementation detail to get the pending operations working and restore the old `run` signature?

Or is there any new use-case that would desire this behavior, and not a full blown completer that also reports exceptions?

Reply:
Yeah sounds good. Then we could also talk about how that API should behave